File: code/c/ds/ll.go
package main

import(
    "fmt"

)

type list struct {
  data int
  next *list
}

/**************

UNLIKE C WHERE WE USE DOT(.) OPERATOR FOR VALUES AND ARROW(->) OPERATOR
FOR POINTERS
GOLANG USES ONLY DOT(.) OPERATOR N IT IDENTIFIES ITSELF ITS VALUE

**************/
func main() {
  //*, *curr list
  //headPtr = new list
  //headPtr -> data = 4
  headPtr := new(list)
  headPtr.data = 4
  headPtr.next = new(list)
  curr := headPtr.next
  curr.data = 5
  curr.next = new(list)
  curr = curr.next
  curr.data = 6
  curr.next = nil

  curr = headPtr

  for i:=0; i<3; i++ {
    fmt.Println(curr.data)
    curr = curr.next
  }


}

File: code/c/ds/ll.cpp
#include <iostream>
using namespace std;

struct list {
  int data;
  list *next;
};


int main() {

    list *head, *curr;
    head = new list;
    head -> data = 5;
    head -> next = new list;
    curr = head->next;
    curr -> data = 7;
    curr -> next = new list ;
    curr = curr -> next;
    curr -> data = 10;
    curr -> next = NULL;

    curr = head;

    for (int i=0; i<3; i++) {
      cout<<curr->data<<endl;
      curr = curr->next;
    }


  return 0;
}
